keps/sig-network/2943-networkpolicy-status/README.md

```diff
@@ -0,0 +1,462 @@
+
+Should be targeted in Beta
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+
+###### How can an operator determine if the feature is in use by workloads?
+
+A new `network_policy_status` metric should be created and the amount 
+of operations executed on this endpoint should be monitored.
+
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] API .status
+  - Condition name: All
+  - Other field: All
+
+The .status field will be created together with the Conditions
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+ - per-day percentage of API calls finishing with 5XX errors <= 1%
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Metrics
+  - Metric name: network_policy_status
+  - Components exposing the metric: APIServer
+- [X] Other 
+  - Details: Network Policy Providers exposing metrics of 
+  communication with APIServer and specifically Status operations
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A
+
+### Dependencies
+
+
+###### Does this feature depend on any specific services running in the cluster?
+
+
+  - [Network Policy Provider]
+    - Usage description: Provide status on policies parsing
+      - Impact of its outage on the feature: No status will be available for users
+      - Impact of its degraded performance or high-error rates on the feature: Inconsistent status may be reported to users
+
+### Scalability
+
+
+###### Will enabling / using this feature result in any new API calls?
+
+  - API call type: GET/POST/PUT/PATCH in networkpolicies/status
+  - estimated throughput: Depends on the amount of existing Network Policies on cluster
+  - originating component(s): Network Policy provider
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes:
+
+  - API type(s): NetworkPolicy / NetworkPolicyStatus
+  - Estimated increase in size: New array of conditions with at least 512 bytes each (Status, Type, Message)
+  - Estimated amount of new objects: New Status field for every existing Network Policy object
+
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+It may get some resource usage increase in disk of Etcd (due to constant patching of status field), and in the 
```

Comment:
and kube-apiserver (which would be more visible, because kube-apiserver is serving watches and all netpol providers agents watch them.